File: MiniDOMFinanziaria/Finanziaria/RichiestaApprovazioneGroup/RichiesteApprovazione/CRichiestaApprovazioneCursor.cs
using System;

namespace minidom
{
    public partial class Finanziaria
    {



        /// <summary>
    /// Cursore sulla tabella degli obiettivi
    /// </summary>
    /// <remarks></remarks>
        public class CRichiestaApprovazioneCursor : Databases.DBObjectCursorPO<CRichiestaApprovazione>
        {
            private DBCursorField<int> m_IDGruppo = new DBCursorField<int>("IDGruppo");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");
            private DBCursorField<int> m_IDOggettoApprovabile = new DBCursorField<int>("IDOggettoApprovabile");
            private DBCursorStringField m_TipoOggettoApprovabile = new DBCursorStringField("TipoOggettoApprovabile");
            private DBCursorField<DateTime> m_DataRichiestaApprovazione = new DBCursorField<DateTime>("DataRichiestaApprovazione");
            private DBCursorField<int> m_IDUtenteRichiestaApprovazione = new DBCursorField<int>("IDUtenteRichiestaApprovazione");
            private DBCursorStringField m_NomeUtenteRichiestaApprovazione = new DBCursorStringField("NomeUtenteRichiestaApprovazione");
            private DBCursorField<int> m_IDMotivoRichiesta = new DBCursorField<int>("IDMotivoRichiesta");
            private DBCursorStringField m_NomeMotivoRichiesta = new DBCursorStringField("NomeMotivoRichiesta");
            private DBCursorStringField m_DescrizioneRichiesta = new DBCursorStringField("DescrizioneRichiesta");
            private DBCursorStringField m_ParametriRichiesta = new DBCursorStringField("ParametriRichiesta");
            private DBCursorField<StatoRichiestaApprovazione> m_StatoRichiesta = new DBCursorField<StatoRichiestaApprovazione>("StatoRichiesta");
            private DBCursorField<DateTime> m_DataPresaInCarico = new DBCursorField<DateTime>("DataPresaInCarico");
            private DBCursorField<int> m_IDPresaInCaricoDa = new DBCursorField<int>("IDPresaInCaricoDa");
            private DBCursorStringField m_NomePresaInCaricoDa = new DBCursorStringField("NomePresaInCaricoDa");
            private DBCursorField<DateTime> m_DataConferma = new DBCursorField<DateTime>("DataConferma");
            private DBCursorField<int> m_IDConfermataDa = new DBCursorField<int>("IDConfermataDa");
            private DBCursorStringField m_NomeConfermataDa = new DBCursorStringField("NomeConfermataDa");
            private DBCursorStringField m_MotivoConferma = new DBCursorStringField("MotivoConferma");
            private DBCursorStringField m_DettaglioConferma = new DBCursorStringField("DettaglioConferma");
            private DBCursorField<int> m_IDCliente = new DBCursorField<int>("IDCliente");
            private DBCursorStringField m_NomeCliente = new DBCursorStringField("NomePersona");

            public CRichiestaApprovazioneCursor()
            {
            }

            public DBCursorField<int> IDGruppo
            {
                get
                {
                    return m_IDGruppo;
                }
            }

            public DBCursorField<int> IDCliente
            {
                get
                {
                    return m_IDCliente;
                }
            }

            public DBCursorStringField NomeCliente
            {
                get
                {
                    return m_NomeCliente;
                }
            }

            public DBCursorField<int> IDOggettoApprovabile
            {
                get
                {
                    return m_IDOggettoApprovabile;
                }
            }

            public DBCursorStringField TipoOggettoApprovabile
            {
                get
                {
                    return m_TipoOggettoApprovabile;
                }
            }

            public DBCursorField<DateTime> DataRichiestaApprovazione
            {
                get
                {
                    return m_DataRichiestaApprovazione;
                }
            }

            public DBCursorField<int> IDUtenteRichiestaApprovazione
            {
                get
                {
                    return m_IDUtenteRichiestaApprovazione;
                }
            }

            public DBCursorStringField NomeUtenteRichiestaApprovazione
            {
                get
                {
                    return m_NomeUtenteRichiestaApprovazione;
                }
            }

            public DBCursorField<int> IDMotivoRichiesta
            {
                get
                {
                    return m_IDMotivoRichiesta;
                }
            }

            public DBCursorStringField NomeMotivoRichiesta
            {
                get
                {
                    return m_NomeMotivoRichiesta;
                }
            }

            public DBCursorStringField DescrizioneRichiesta
            {
                get
                {
                    return m_DescrizioneRichiesta;
                }
            }

            public DBCursorStringField ParametriRichiesta
            {
                get
                {
                    return m_ParametriRichiesta;
                }
            }

            public DBCursorField<StatoRichiestaApprovazione> StatoRichiesta
            {
                get
                {
                    return m_StatoRichiesta;
                }
            }

            public DBCursorField<DateTime> DataPresaInCarico
            {
                get
                {
                    return m_DataPresaInCarico;
                }
            }

            public DBCursorField<int> IDPresaInCaricoDa
            {
                get
                {
                    return m_IDPresaInCaricoDa;
                }
            }

            public DBCursorStringField NomePresaInCaricoDa
            {
                get
                {
                    return m_NomePresaInCaricoDa;
                }
            }

            public DBCursorField<DateTime> DataConferma
            {
                get
                {
                    return m_DataConferma;
                }
            }

            public DBCursorField<int> IDConfermataDa
            {
                get
                {
                    return m_IDConfermataDa;
                }
            }

            public DBCursorStringField NomeConfermataDa
            {
                get
                {
                    return m_NomeConfermataDa;
                }
            }

            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            public DBCursorStringField MotivoConferma
            {
                get
                {
                    return m_MotivoConferma;
                }
            }

            public DBCursorStringField DettaglioConferma
            {
                get
                {
                    return m_DettaglioConferma;
                }
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Database;
            }

            protected override Sistema.CModule GetModule()
            {
                return RichiesteApprovazione.Module;
            }

            public override string GetTableName()
            {
                return "tbl_CQSPDRichiesteApprovazione";
            }
        }
    }
}

File: MiniDOMFinanziaria/Finanziaria/RichiestaApprovazioneGroup/RichiesteApprovazione/CRichiestaApprovazioneCursor.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica


Partial Public Class Finanziaria

     

    ''' <summary>
    ''' Cursore sulla tabella degli obiettivi
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CRichiestaApprovazioneCursor
        Inherits DBObjectCursorPO(Of CRichiestaApprovazione)

        Private m_IDGruppo As New CCursorField(Of Integer)("IDGruppo")
        Private m_Flags As New CCursorField(Of Integer)("Flags")
        Private m_IDOggettoApprovabile As New CCursorField(Of Integer)("IDOggettoApprovabile")
        Private m_TipoOggettoApprovabile As New CCursorFieldObj(Of String)("TipoOggettoApprovabile")
        Private m_DataRichiestaApprovazione As New CCursorField(Of Date)("DataRichiestaApprovazione")
        Private m_IDUtenteRichiestaApprovazione As New CCursorField(Of Integer)("IDUtenteRichiestaApprovazione")
        Private m_NomeUtenteRichiestaApprovazione As New CCursorFieldObj(Of String)("NomeUtenteRichiestaApprovazione")
        Private m_IDMotivoRichiesta As New CCursorField(Of Integer)("IDMotivoRichiesta")
        Private m_NomeMotivoRichiesta As New CCursorFieldObj(Of String)("NomeMotivoRichiesta")
        Private m_DescrizioneRichiesta As New CCursorFieldObj(Of String)("DescrizioneRichiesta")
        Private m_ParametriRichiesta As New CCursorFieldObj(Of String)("ParametriRichiesta")
        Private m_StatoRichiesta As New CCursorField(Of StatoRichiestaApprovazione)("StatoRichiesta")
        Private m_DataPresaInCarico As New CCursorField(Of Date)("DataPresaInCarico")
        Private m_IDPresaInCaricoDa As New CCursorField(Of Integer)("IDPresaInCaricoDa")
        Private m_NomePresaInCaricoDa As New CCursorFieldObj(Of String)("NomePresaInCaricoDa")

        Private m_DataConferma As New CCursorField(Of Date)("DataConferma")
        Private m_IDConfermataDa As New CCursorField(Of Integer)("IDConfermataDa")
        Private m_NomeConfermataDa As New CCursorFieldObj(Of String)("NomeConfermataDa")

        Private m_MotivoConferma As New CCursorFieldObj(Of String)("MotivoConferma")
        Private m_DettaglioConferma As New CCursorFieldObj(Of String)("DettaglioConferma")

        Private m_IDCliente As New CCursorField(Of Integer)("IDCliente")
        Private m_NomeCliente As New CCursorFieldObj(Of String)("NomePersona")

        Public Sub New()
        End Sub

        Public ReadOnly Property IDGruppo As CCursorField(Of Integer)
            Get
                Return Me.m_IDGruppo
            End Get
        End Property

        Public ReadOnly Property IDCliente As CCursorField(Of Integer)
            Get
                Return Me.m_IDCliente
            End Get
        End Property

        Public ReadOnly Property NomeCliente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCliente
            End Get
        End Property


        Public ReadOnly Property IDOggettoApprovabile As CCursorField(Of Integer)
            Get
                Return Me.m_IDOggettoApprovabile
            End Get
        End Property

        Public ReadOnly Property TipoOggettoApprovabile As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoOggettoApprovabile
            End Get
        End Property

        Public ReadOnly Property DataRichiestaApprovazione As CCursorField(Of Date)
            Get
                Return Me.m_DataRichiestaApprovazione
            End Get
        End Property

        Public ReadOnly Property IDUtenteRichiestaApprovazione As CCursorField(Of Integer)
            Get
                Return Me.m_IDUtenteRichiestaApprovazione
            End Get
        End Property

        Public ReadOnly Property NomeUtenteRichiestaApprovazione As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeUtenteRichiestaApprovazione
            End Get
        End Property

        Public ReadOnly Property IDMotivoRichiesta As CCursorField(Of Integer)
            Get
                Return Me.m_IDMotivoRichiesta
            End Get
        End Property

        Public ReadOnly Property NomeMotivoRichiesta As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeMotivoRichiesta
            End Get
        End Property

        Public ReadOnly Property DescrizioneRichiesta As CCursorFieldObj(Of String)
            Get
                Return Me.m_DescrizioneRichiesta
            End Get
        End Property

        Public ReadOnly Property ParametriRichiesta As CCursorFieldObj(Of String)
            Get
                Return Me.m_ParametriRichiesta
            End Get
        End Property

        Public ReadOnly Property StatoRichiesta As CCursorField(Of StatoRichiestaApprovazione)
            Get
                Return Me.m_StatoRichiesta
            End Get
        End Property

        Public ReadOnly Property DataPresaInCarico As CCursorField(Of Date)
            Get
                Return Me.m_DataPresaInCarico
            End Get
        End Property

        Public ReadOnly Property IDPresaInCaricoDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDPresaInCaricoDa
            End Get
        End Property

        Public ReadOnly Property NomePresaInCaricoDa As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomePresaInCaricoDa
            End Get
        End Property

        Public ReadOnly Property DataConferma As CCursorField(Of Date)
            Get
                Return Me.m_DataConferma
            End Get
        End Property

        Public ReadOnly Property IDConfermataDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDConfermataDa
            End Get
        End Property

        Public ReadOnly Property NomeConfermataDa As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeConfermataDa
            End Get
        End Property


        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property MotivoConferma As CCursorFieldObj(Of String)
            Get
                Return Me.m_MotivoConferma
            End Get
        End Property

        Public ReadOnly Property DettaglioConferma As CCursorFieldObj(Of String)
            Get
                Return Me.m_DettaglioConferma
            End Get
        End Property


        Protected Overrides Function GetConnection() As CDBConnection
            Return Finanziaria.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Finanziaria.RichiesteApprovazione.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_CQSPDRichiesteApprovazione"
        End Function
    End Class




End Class
